# U.S. Medical Insurance Costs

In [15]:
# Goal of this project is to find the cost difference between smoker and non-smokers, does it differ per region?
# Average cost per individual in each region
# the region with the highest BMI
import csv
import pandas as pd

In [16]:
# variables for all the information in the csv file
age = []
sex = []
BMI = []
children = []
smoker = []
region = []
charges = []

In [17]:
insurance_data = []
with open('insurance.csv', newline = '') as insurance:
    insurance_info = csv.DictReader(insurance)
    for row in insurance_info:
        age.append(row['age'])
        sex.append(row['sex'])
        BMI.append(row['bmi'])
        children.append(row['children'])
        smoker.append(row['smoker'])
        region.append(row['region'])
        charges.append(row['charges'])
  
  
            
      

In [18]:
 # Average cost per individual
 def average_cost_total(charges):
    total_charges = 0
    for cost in charges:
        cost = float(cost)
        total_charges += cost
    average_cost = total_charges/len(charges)
    return int(average_cost)
print("The average cost per individual is " + str(average_cost_total(charges)))  

The average cost per individual is 13270


The average cost per individual is 13270.

In [19]:
df = pd.read_csv("insurance.csv")
average_charges_by_region = df.groupby("region")["charges"].mean()
print(average_charges_by_region)


region
northeast    13406.384516
northwest    12417.575374
southeast    14735.411438
southwest    12346.937377
Name: charges, dtype: float64


The Average cost per region is:
Northeast = 13406.38
Northwest = 12417.58
Southeast = 14735.41
Southwest = 12346.94
The Southeast has the highest average insurance cost.

In [20]:
df = pd.read_csv("insurance.csv")
average_BMI_by_region = df.groupby("region")["bmi"].mean()
print(average_BMI_by_region)

region
northeast    29.173503
northwest    29.199785
southeast    33.355989
southwest    30.596615
Name: bmi, dtype: float64


When looking at the average BMI per region we notice that the Southeast has the highest BMI and that results in the higher cost per average. 

In [21]:
df = pd.read_csv("insurance.csv")
average_charges_by_smoker = df.groupby("smoker")["charges"].mean()
print(average_charges_by_smoker)

smoker
no      8434.268298
yes    32050.231832
Name: charges, dtype: float64


When looking at the cost difference of insurance of smoker vs non smoker the smoker has a much higher cost of insurance. The difference of smoker and non smoker is 23,615.96. That is a huge difference in price of insurance.